src/SoCreate.ServiceFabric.PubSub/State/ReferenceWrapper.cs

```diff
@@ -52,12 +57,14 @@ protected IHashingHelper HashingHelper
         /// <summary>
         /// Creates a new instance with an optional routing key.
         /// </summary>
-        /// <param name="routingKey">Optional routing key to filter messages based on content. 'Key=Value' where Key is a message property path and Value is the value to match with message payload content.</param>
-        protected ReferenceWrapper(string routingKey = null)
+        /// <param name="routingKeyName">Optional Message property path used for routing</param>
+        /// <param name="routingKeyValue">Optional value to match with message payload content used for routing.</param>
+        protected ReferenceWrapper(string routingKeyName = null, string routingKeyValue = null)
```

Comment:
Revert this to accept a single parameter 'routingKey'.  Split on '=' and validate as it was doing before.  Then assign `RoutingKeyName` and `RoutingKeyValue` instead of `RoutingKey`.

Reply:
If you want to go the regex route should we still have a ```RoutingKeyValue``` property that is a string or should we just create the regex in the constructor?